meson.build

```diff
@@ -50,6 +50,26 @@ add_project_arguments(
   '-I' + meson.build_root(),
   language: 'c', native: true)
 
+# The following flags are applied only to cross builds
+add_project_arguments(
+  '-nostartfiles', '-nostdlib',
+  '-Os', '-g',
+  '-static',
+  '-fvisibility=hidden',
+  language: 'cpp', native: false)
+add_project_arguments(
+  '-nostartfiles', '-nostdlib',
+  '-Os', '-g',
+  '-static',
+  '-fvisibility=hidden',
+  language: 'c', native: false)
+add_project_link_arguments(
+  '-nostdlib',
+  language: 'cpp', native: false)
+add_project_link_arguments(
+  '-nostdlib',
+  language: 'c', native: false)
+
```

Comment:
I think this is an excellent opportunity to add comments explaining what all these are for, and why we need to include them. IMO I think it's important to justify unusual compiler flags. -Os is obvious, I guess, but -fvisibility, somewhat less so.

Reply:
My aim with this commit was just to move the arguments between files. I'm happy to do follow-up commits in this PR though, if they are not controversial.

I was thinking about a follow-up PR which would create particular argument groups for different use-cases. For instance, I know it would be useful to have a `-O3` argument set which can be used for CoreMark and maybe other benchmarking. This is different to `-Oz` (and other flags) which we want for compiling some device software, and also different to passing `-g` (and other flags) when someone is trying to debug their software.

Reply:
Sure, I think those would be good followups; I think this is otherwise fine to submit as-is.